Assets/HoloToolkit/Common/Scripts/Singleton.cs

```diff
@@ -6,12 +6,13 @@
 namespace HoloToolkit.Unity
 {
     /// <summary>
-    /// Singleton behaviour class, used for components that should only have one instance.
+    /// Singleton behavior class, used for components that should only have one instance.
     /// <remarks>Singleton classes live on through scene transitions and will mark their 
     /// parent root GameObject with <see cref="Object.DontDestroyOnLoad"/></remarks>
     /// </summary>
     /// <typeparam name="T">The Singleton Type</typeparam>
-    public class Singleton<T> : MonoBehaviour where T : Singleton<T>
+    [DisallowMultipleComponent]
```

Comment:
Great addition!

Reply:
I tried this out but it doesn't actually seem to do anything in the editor.

I was able to add a gaze manager component multiple times anyway.

Reply:
Sounds like a bug report for unity